src/main/java/net/minecraftforge/common/config/Configuration.java

```diff
@@ -80,9 +89,16 @@
 
     public Configuration(){}
 
+    @Deprecated
+    public Configuration(CharSource staticSource, CharSink staticSink){
```

Comment:
Mostly server -> client synchornization.
For example: on server you write configuration to ByteBufOutputStream and send the IMessage. Then on client, you load to your configuration using ByteBufInputStream from IMessage.
Currently, if you want to sync configuration, you either write your own configuration API (or use another one, like jsons), or sync by using temp files (configuration->tmp->bytebuf->_NET_->bytebuf->tmp->configuration).